docker/Dockerfile-focal

```diff
@@ -19,11 +33,16 @@ RUN apt-get update && apt-get install -yq \
 
 WORKDIR /opt/conda_setup
 
-RUN curl -o miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
-     chmod +x miniconda.sh && \
-     ./miniconda.sh -b -p /opt/conda && \
-     /opt/conda/bin/conda install -y python=$PYTHON_VERSION && \
-     /opt/conda/bin/conda clean -ya
+RUN case ${TARGETPLATFORM} in \
```

Comment:
These days, I'd probably use uv and nothing else but we don't need to change this in this PR.

Reply:
That would probably help save GitHub minutes too as the non-nethack parts are likely the majority of the build time.

Reply:
What's uv?

Reply:
https://astral.sh/blog/uv

I'm already using it in one of the tests, [here](https://github.com/heiner/nle/blob/main/.github/workflows/lint_python.yml#L18).

It's from the `ruff` developer and one of the many Rust-for-Python tools that massively speed up things these days.

Reply:
On my M3 Mac, the Docker build takes 15 minutes and the image size is a chunky 23GB.

Reply:
Is it faster the second time and we just have to cache the Docker layers?

Alternatively, I think the base image with CUDA is quite heavy.

Reply:
It is faster, but still takes nearly 10 minutes, most of the time is in `RUN pip install '.[all]'` on amd64 (~400s). Rather strangely, the same step for arm64 only takes about 40s. I wonder is this due to me running the build on a Mac.

Reply:
Yes, that's likely. Cross compilation is rather slow on arm MacBooks.